### ISSUE TITLE
Organization-Level environmental variables

### ISSUE THREAD

**Issue by bpeake-illuscio** [readthedocs/readthedocs.org#6043]
Hello,

I use a pypi repository hosted through azure artifacts for my company's python packages. As such, to build documentation, I need to pass environmental variables: PIP_INDEX_URL and PIP_EXTRA_INDEX_URL to EVERY project in my organization, and update EVERY project when the password changes every few months for security reasons.

It would be awesome if I could set an organization-wide variable once and have it apply to every project I build documentation for.

This will also be a little less of a pain once the REST API V3 becomes available for the public site as I can at least automate updating each project's environmental variables.

Thanks!

**Comment by stsewd** [comment]
Hi @bpeake-illuscio. With "every project in my organization" do you mean organizations from readthedocs.com or from an account in readthedocs.org?

> This will also be a little less of a pain once the REST API V3 becomes available for the public site as I can at least automate updating each project's environmental variables.

Yeah, I think that would be a good solution. We don't have something like a _global setting for all projects_. So not sure how having global env variables would fit.

**Comment by bpeake-illuscio** [comment]
Sorry, I guess I meant organizations from readthedocs.com. I have about 20 libraries which all need the same variables changed regularly, and that number is only going to grow.

Thanks!

**Comment by flying-sheep** [comment]
We need that too. GitHub’s org secrets are a blessing and much nicer than manually doing things.

I guess I could maybe write automation that modifies each project on creation (can I?), but a setting in the GUI would be much better.

**Comment by bn-andrew** [comment]
Yeah like above, we could use this immensely. We have ~60 projects currently (all sub projects of one if that helps) and setting the env vars for each is tedious.
Having a setting at the root project and / or the organisation admin that acts as a default (any project specific entries would just overrule) would be super helpful.

Alternatively it would be great if there was some sort of actual secret store, either internal or external eg AWS Secrets.
Because we would also really like to use this on our PR builds (everything is private though)

Thanks
